### ISSUE TITLE
alternate repository ?

### ISSUE THREAD

**Issue by bomilkar** [XCSoar/xcsoar-data-repository#171]
Although it's an Issue for XCSoar I reckon the audience here has a vested interest in this:

Currently the URI of the "repository" is hard coded in
_src/Repository/Glue.cpp_
It points to "http://download.xcsoar.org/repository".

This "one size fits all" approach is probably not ideal for several reasons: I might never be interested in data for the opposite side of the globe. I might want to publish the waypoints for a local competition which in turn is pointless for everybody else. etc.

Are there any plans to allow users to configure their own repository URI?

**Comment by lordfolken** [comment]
Hello there, 
I also don't like hardcoded defaults, especially for resources that are external. So yes, the url should be configurable, or even extensible in the style of apt linux distro package repositories. 

For waypoint files and other data there is the waypoints-special folder. You can use this for your local competition if you'd like. The data entry is just a few bytes and doesn't really matter to anyone downloading it. I hope that by being able to submit pull requests and reviews by end users that at least some would add their local data there. 

Competition data at this point however seems to be dominated by https://soaringspot.com. While I'm not a fan of driving users to commercial services, its at least free to use. They have an API, which allows access by 3rd party apps. 

@kedder has developed a cli gui for openvario to interface and download competition data from soaringspot: https://github.com/kedder/openvario-compman maybe this can be extended for use on other platforms.

**Comment by bomilkar** [comment]
@lordfolken what do you mean by 

> For waypoint files and other data there is the waypoints-special folder.

Can you elaborate a bit how that would work?

I'm glad you agree the url should be configurable.
I can think of 2 ways:
1. put a parameter in Config->System->Setup. Maybe in "XCSoar Cloud"?
2. make it a command line argument "--repository=http://download.xcsoar.org/repository".
Anything else?
Any preferences?

**Comment by lordfolken** [comment]
what i meant with waypoints which are not country specific, ie. competition or otherwise:
https://github.com/XCSoar/xcsoar-data-repository/blob/master/data/waypoints-special.json

I'd put it under setup -> repository in expert mode.

**Comment by lordfolken** [comment]
to be completely honest, i'd still prefer if people would contribute to the official repository, than try to circumvent the polices the project has put in place by providing their own repos.

**Comment by bomilkar** [comment]
> what i meant with waypoints which are not country specific, ie. competition or otherwise:
> https://github.com/XCSoar/xcsoar-data-repository/blob/master/data/waypoints-special.json

I still don't see how I can access these listed 4 items. I don't see them as part of the (long) list of waypoints.

**Comment by bomilkar** [comment]
> to be completely honest, i'd still prefer if people would contribute to the official repository, than try to circumvent the polices the project has put in place by providing their own repos.

I agree with you but I have a number of issues with what's currently on the list of Downloads:
1. I know nothing about these waipoints or airspaces: when have they been updated, what is the change history, can I trust it at all? Checking all this is pointless when you're sitting in the glider pressing the Download button. But more information which I can investigate at home would be needed.
2. I don't want to switch waipoints or airspaces every time I cross county borders. I want waipoints and airspaces to cover (at least) 500km radius from my takeoff location. In theory, I could concatenate several files into one but that's not supported.
3. I would also expect some sort of directory structure or select criteria by which I can shorten the list to the region or purpose I'm interested in. Instead of being presented with a list of 30+ files from around the globe.

The above mentioned led me to create my own maps, waipoints and airspaces.
I derive waipoints and airspaces from OpenAIP's XML sources which I have found to be very reliable at least for my region. They track changes and they are very responsive to update their data if need be. While I can use this for my own use and probably share it with my friends I don't feel entitled by OpenAIP to share it with "the community".
When I fly from my home base I use maps, waipoints and airspaces to cover "Central Europe". This might have been a challenge for the old PDAs but it's not a problem at all for nowadays devices.

So the bottom line from my perspective: it takes a big and continuous effort to make the current Download feature very useful for the community. In the meantime we can use a workaround implementing "private" repos.

**Comment by lordfolken** [comment]
1. XCSoar should never be your primary navigation. That's why you are required to carry maps. Its a convenience at best. The sources for the files are not trustworthy, nor is the update process guaranteed. None of the FIRs publish their airspace data openly. And those that where approached refused and hid behind legal and commercial reasons. That an organization that boasts the slogan "one single sky" is not capable of publishing airspace data in a single format for all Europe, is beyond me. 


2. I understand this. XCsoar allows for an additional airspace file to be loaded in xpert mode. But of course this doesn't deduplicate airspaces. Also i noticed that coordinates of air-spaces hardly match between files, which would make de duplication very difficult and error prone. 

3. XCsoar would have an iso code filter in the repository, but nobody implemented this yet.

**Comment by bomilkar** [comment]
Yes, it's really sad that we still **have to** use paper maps for navigation. Therefore the legal disclaimer is redundant although I would recommend to use a similar statement along side of XCSoar's repository.
However, (legal) ICAO maps are not very helpful at times. Examples: try to cross the Inn valley near Landeck. Or pass by the Munich airspace near the north end of the Ammersee. Airspace structures can be **very** convoluted and hard to visualize from looking at the map.
And we all know ICAO maps are far from perfect. They list airfields with all the attributes even though they were given up years ago. A lot of UL airfields aren't even listed on ICAO maps, yet they make perfect outlanding spots. But all this is a secondary issue.

I think if we offer a repo we should do our best to keep the content correct. This may require different approaches in different parts of the world. I can only comment on Central Europe and as I stated before I found OpenAIP to be a very good staring point. It's much better than starting from scratch or with a database that hasn't been maintained for years. In addition, yesterday I have received an email from Stephan Besser from OpenAIP stating that OpenAIP encourages the XCSoar community to leverage their database as long as the source is mentioned and it is distributed free of charge. Of course, he would like to see content changes being made to the source if at all possible. (I can share this email, in German.) 
De duplication can be a bit tricky for airspaces in particular. But S/W can do it ;-) and duplicates aren't a real issue.
@lordfolken can you point me to the "iso code filter" you had mentioned in your post?
